### ISSUE TITLE
The returns of DataPipeline.sarsd_iter() are incorrect sometimes

### ISSUE THREAD

**Issue by fA1sEr** [minerllabs/minerl#157]
I run ```test.py``` repeatedly
```
import minerl

data = minerl.data.make('MineRLObtainDiamond-v0', data_dir='/home/lzhang/minerl_data/')

for current_state, action, reward, next_state, done in \
    data.sarsd_iter(num_epochs=1, max_sequence_len=32):
    print(current_state['inventory']['iron_ore'])
    break
```
The output shape is not the same every time
```
(venv) lzhang@cs1029-workstation:~$ python test.py 
0
(venv) lzhang@cs1029-workstation:~$ python test.py 
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
(venv) lzhang@cs1029-workstation:~$ python test.py 
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
(venv) lzhang@cs1029-workstation:~$ python test.py 
0
(venv) lzhang@cs1029-workstation:~$ python test.py 
0
(venv) lzhang@cs1029-workstation:~$ python test.py 
0
(venv) lzhang@cs1029-workstation:~$ python test.py 
0
(venv) lzhang@cs1029-workstation:~$ python test.py 
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
(venv) lzhang@cs1029-workstation:~$ python test.py 
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
(venv) lzhang@cs1029-workstation:~$ python test.py 
0
```
And ```action['camera']``` ```action['forward']``` ```current_state['equipped_items']['mainhand']['damage']``` may also be incorrect.

**Comment by fA1sEr** [comment]
The version of minerl is ```0.2.2```
and the dataset is downloaded from ```https://router.sneakywines.me/minerl_v1/data_texture_0_low_res.tar.gz```

**Comment by fA1sEr** [comment]
And it will crash sometimes
```
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    data.sarsd_iter(num_epochs=1, max_sequence_len=32):
  File "/home/lzhang/venv/lib/python3.5/site-packages/minerl/data/data_pipeline.py", line 193, in sarsd_iter
    observation_dict = DataPipeline.map_to_dict(observation_seq, gym_spec._kwargs['observation_space'])
  File "/home/lzhang/venv/lib/python3.5/site-packages/minerl/data/data_pipeline.py", line 118, in map_to_dict
    index = _map_to_dict(index, handler_list, key, space, result)
  File "/home/lzhang/venv/lib/python3.5/site-packages/minerl/data/data_pipeline.py", line 106, in _map_to_dict
    _map_to_dict(idx, src[i].T, k, s, inner_dict)
  File "/home/lzhang/venv/lib/python3.5/site-packages/minerl/data/data_pipeline.py", line 113, in _map_to_dict
    dst[key] = src[i]
IndexError: index 5 is out of bounds for axis 0 with size 5
```

**Comment by minerllabs** [comment]
are you using python 3.5?

**Comment by MadcowD** [comment]
@brandonhoughton, lets try implementing ordered dicts asap

MineRL Labs <notifications@github.com> schrieb am Mo. 22. Juli 2019 um
10:31:

**Comment by fA1sEr** [comment]
> are you using python 3.5?

yes

**Comment by MadcowD** [comment]
#137

**Comment by MadcowD** [comment]
We're going to fix this bug for you. We're on it rn!

**Comment by minerllabs** [comment]
fixed in #184
